adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -21,6 +32,98 @@ table below.
 
 == Usage example
 
+The example below rewrites the application from
+<<subsec:example.sycl.application>> to demonstrate the usage of this extension.
+
+[source,role=synopsis]
+----
+#include <iostream>
+#include <sycl/sycl.hpp>
+using namespace sycl;  // (optional) avoids need for "sycl::" before SYCL names
+
+int main() {
+  // Create a queue to work on.
+  queue myQueue;
+  
+  khr::command_graph myGraph{myQueue};
+
+  // Create some 2D buffers of float for our matrices
+  buffer<float, 2> a{range<2>{N, M},
+                     khr::property::graph::assume_buffer_outlives_graph{}};
+  buffer<float, 2> b{range<2>{N, M},
+                     khr::property::graph::assume_buffer_outlives_graph{}};
+  buffer<float, 2> c{range<2>{N, M},
+                     khr::property::graph::assume_buffer_outlives_graph{}};
+
+  myGraph.begin_recording(myQueue);
+
+  // Launch an asynchronous kernel to initialize a
+  myQueue.submit([&](handler& cgh) {
+    // The kernel writes a, so get a write accessor on it
+    accessor A{a, cgh, write_only};
+
+    // Enqueue a parallel kernel iterating on a N*M 2D iteration space
+    cgh.parallel_for(range<2>{N, M},
+                     [=](id<2> index) { A[index] = index[0] * 2 + index[1]; });
+  });
+
+  // Launch an asynchronous kernel to initialize b
+  myQueue.submit([&](handler& cgh) {
+    // The kernel writes b, so get a write accessor on it
+    accessor B{b, cgh, write_only};
+
+    // From the access pattern above, the SYCL runtime detects that this
+    // command_group is independent from the first one and can be
+    // scheduled independently
+
+    // Enqueue a parallel kernel iterating on a N*M 2D iteration space
+    cgh.parallel_for(range<2>{N, M}, [=](id<2> index) {
+      B[index] = index[0] * 2014 + index[1] * 42;
+    });
+  });
+
+  // Launch an asynchronous kernel to compute matrix addition c = a + b
+  myQueue.submit([&](handler& cgh) {
+    // In the kernel a and b are read, but c is written
+    accessor A{a, cgh, read_only};
+    accessor B{b, cgh, read_only};
+    accessor C{c, cgh, write_only};
+
+    // From these accessors, the SYCL runtime will ensure that when
+    // this kernel is run, the kernels computing a and b have completed
+
+    // Enqueue a parallel kernel iterating on a N*M 2D iteration space
+    cgh.parallel_for(range<2>{N, M},
+                     [=](id<2> index) { C[index] = A[index] + B[index]; });
+  });
+  
+  myGraph.end_recording();
+  
+  auto myExecGraph = myGraph.finalize();
```

Comment:
Should we avoid using auto here and explicitly show the `command_graph<graph_state::executable>`? Since we are detailing the extension and its API I think that would be clearer.